allennlp/data/samplers/bucket_batch_sampler.py

```diff
@@ -92,28 +100,41 @@ def _argsort_by_padding(self, instances: Iterable[Instance]) -> List[int]:
         instances_with_lengths = []
         for instance in instances:
             # Make sure instance is indexed before calling .get_padding
+            true_lengths = []
             lengths = []
             for field_name in self.sorting_keys:
                 if field_name not in instance.fields:
                     raise ConfigurationError(
                         f'Sorting key "{field_name}" is not a field in instance. '
                         f"Available fields/keys are {list(instance.fields.keys())}."
                     )
-                lengths.append(
-                    add_noise_to_value(len(instance.fields[field_name]), self.padding_noise)
-                )
-            instances_with_lengths.append((lengths, instance))
+                true_lengths.append(len(instance.fields[field_name]))
+
+                lengths.append(add_noise_to_value(true_lengths[-1], self.padding_noise))
```

Comment:
It's a small thing, but I'd rather `lengths` was called `noisy_length`, and the other one just `length`.